### DISCUSSION TITLE
Best approach to separate authenticated users from guest users

### DISCUSSION THREAD

**Discussion by de-raaf-media** [duncanmcclean/simple-commerce#1177] [Help] [answered]
Hi there,

What would be the best way to separate logged in users from guest users? Because in a users table an email address should be unique. I currently use the users table to store the authenticated users. Would a separate column for the guest users e-mail address be a good idea for this? Or maybe not even store the customer in a separate table at all but only in the order itself?

**Comment by duncanmcclean** [comment]
When you say "authenticated users" and "guest users", what do you mean.

Where are authenticated users authenticating - in the Control Panel?

**Reply by de-raaf-media** [reply]
No I mean in the front-end. In my application, a customer can register for an account. But he can also check out as a guest. For both scenarios they are stored in the users table and attached to the order via the customer_id. However for guests this should probably not be persisted in the users table? For instance if a guest user in the future will create an account. The email address is already taken. Maybe I should check if the email address exists in the guest form (unique email required) but I don't know if that is the way to go because then you can see if a user exists in the database by typing it into the form. Maybe there is another way to separate this? A customers table for guests and a users table for registered users?

**Reply by duncanmcclean** [reply, marked as answer]
I'm assuming you're using the `UserCustomerRepository`? I don't think there's a way right now for some users to be stored in one place, with others being stored elsewhere. 🤔

In the next version of Simple Commerce, all customers will be stored as users, with the option to have guest customers who just live on the order. Guest orders will then be converted to user orders if a user registers for an account.

**Reply by de-raaf-media** [reply]
Great news that guest users are supported in the next version. I think we will wait for that then. As our webshop is still in development.
